extensions-core/lookups-cached-global/src/main/java/io/druid/query/lookup/NamespaceLookupExtractorFactory.java

```diff
@@ -95,7 +96,12 @@ public NamespaceLookupExtractorFactory(
   public boolean start()
```

Comment:
If callers need to know, then either adding InterruptedException to throws, or setting `Thread.currentThread.interrupt()` works. If they don't need to know, I think neither of them is really necessary. And the flag can actually be counterproductive if callers aren't checking it, since that can cause the flag to get set on some random thread accidentally.

unfortunately I'm not familiar enough with the lookup code to know if callers need to know about interrupt status…

If you're comfortable with treating interrupts the same as any other arbitrary exception, I think it's fine to keep things the way they are in your diff.